src/main/java/io/github/haoyangw/rica/task/Deadline.java

```diff
@@ -0,0 +1,112 @@
+package io.github.haoyangw.rica.task;
+
+import io.github.haoyangw.rica.exception.RicaSerializationException;
+import io.github.haoyangw.rica.exception.RicaTaskException;
+
+public class Deadline extends Todo {
+    private static final String COMMAND = "deadline";
+    private static final String DEADLINE_KEYWORD = "/by";
+    private static final String MISSING_DEADLINE_ERROR = " Forget something? (Hint: Where's the deadline for this deadline task?)";
+    private static final String MISSING_DESCRIPTION_ERROR = " Did you give a description for this deadline task xD";
+    private static final int NUM_OF_SERIALIZED_DATA = 4;
+    private static final String WRONG_CREATE_CMD_ERROR = " Hmm I'm being told to set a deadline with the wrong command. Help xP";
+    protected static final String TYPE = "D";
+    protected final String deadline;
+
+    public Deadline(String description, String deadline) {
+        this(description, false, deadline);
+    }
+
+    private Deadline(String description, boolean isDone, String deadline) {
+        super(description, isDone);
+        this.deadline = deadline;
+    }
+
+    private static String getDescriptionOf(String command) throws RicaTaskException {
+        String[] parameters = command.split(" ");
+        int MISSING_PARAMS = 1;
+        if (parameters.length == MISSING_PARAMS) {
+            throw new RicaTaskException(Deadline.MISSING_DESCRIPTION_ERROR);
+        }
+        StringBuilder descBuilder = new StringBuilder();
+        for (int i = 1; i < parameters.length; i += 1) {
+            if (parameters[i].equals(Deadline.DEADLINE_KEYWORD)) {
+                if (i == 1) {
+                    throw new RicaTaskException(Deadline.MISSING_DESCRIPTION_ERROR);
+                }
+                break;
+            }
+            if (i != 1) {
```

Comment:
Try to avoid magic literals. There has been a few more instances in the rest of your code. You can use a named constant makes the code easier to understand because the name tells us more about the meaning of the number.